addons/mass_mailing/models/mass_mailing.py

```diff
@@ -620,6 +630,26 @@ def onchange_mass_mailing_campaign_id(self, cr, uid, id, mass_mailing_campaign_i
                    'medium_id': mass_mailing_campaign[0].medium_id.id}
             return {'value': dic}
 
+    def create(self, cr, uid, vals, context=None):
+        if not vals['mass_mailing_campaign_id']:
+            utm_campaign_id = self.pool['utm.campaign'].create(cr, uid, {'name': vals['name']}, context=context)
```

Comment:
I would rely on the default utm_campaign creation from the mass_mailing_campaign model.